pep-0712.rst

```diff
@@ -0,0 +1,267 @@
+each overload's unary argument type. If no overloads are valid, it is a type
+error.
+
+Example
+'''''''
+
+.. code-block:: python
+
+  # The following are valid converter types, with a comment containing the
+  # synthesized __init__ argument's type.
+  converter: Any  # Any
+  def converter(x: int): ...  # int
+  def converter(x: int | str): ...  # int | str
+  def converter(x: int, y: str = "a"): ...  # int
+  def converter(x: int, *args: str): ...  # int
+  def converter(*args: str): ...  # str
+  def converter(*args: str, x: int = 0): ...  # str
+
+  @overload
+  def converter(x: int): ...  # <- valid
+  @overload
+  def converter(x: int, y: str): ...  # <- ignored
+  @overload
+  def converter(x: list): ... # <- valid
+  def converter(x, y = ...): ... # int | list
+
+  # The following are valid converter types for a field annotated as type 'list'.
+  def converter(x) -> list: ...
+  def converter(x) -> Any: ...
+  def converter(x) -> list[int]: ...
+
+  @overload
+  def converter(x: int) -> tuple: ... # <- ignored
+  @overload
+  def converter(x: str) -> list: ... # <- valid
+  @overload
+  def converter(x: bytes) -> list: ... # <- valid
+  def converter(x): ... # __init__ would use argument type 'str | bytes'.
+
+  # The following are invalid converter types.
+  def converter(): ...
+  def converter(**kwargs): ...
+  def converter(x, y): ...
+  def converter(*, x): ...
+  def converter(*args, x): ...
+
+  @overload
+  def converter(): ...
+  @overload
+  def converter(x: int, y: str): ...
+  def converter(x=..., y = ...): ...
+
+  # The following are invalid converter types for a field annotated as type 'list'.
+  def converter(x) -> tuple: ...
+  def converter(x) -> Sequence: ...
+
+  @overload
+  def converter(x) -> tuple: ...
+  @overload
+  def converter(x: int, y: str) -> list: ...
+  def converter(x=..., y = ...): ...
+
+
+Reference Implementation
+========================
+
+The `attrs <#attrs-converters>`_ library already includes a ``converter``
+parameter containing converter semantics.
+
+CPython support can be seen on a branch: `GitHub <#attrs-converters>`_.
+
+Rejected Ideas
+==============
+
+Just adding "converter" to ``dataclass_transform``'s ``field_specifiers``
+-------------------------------------------------------------------------
+
+The idea of isolating this addition to ``dataclass_transform`` was briefly
+discussed in `Typing-sig <#only-dataclass-transform>`_ where it was suggested
+to open this to ``dataclasses``.
+
+Additionally, adding this to ``dataclasses`` ensures anyone can reap the
+benefits without requiring additional libraries.
+
+Automatic conversion using the field's type
+-------------------------------------------
+
+One idea could be to allow the type of the field specified (e.g. ``str`` or
+``int``) to be used as a converter for each argument provided.
+`Pydantic's data conversion <#pydantic-data-conversion>`_ has semantics which
```

Comment:
```suggestion
`Pydantic's data conversion <pydantic-data-conversion_>`__ has semantics which
```

* Fix broken link syntax (as above)
* Remove redundant and confusing `#` in link target (as above)